frontend/api/auth/index.ts

```diff
@@ -0,0 +1,243 @@
+    if (!email) {
+      // This is for facebook user, which doesn't have email
+      try {
+        const user = await prisma.user.upsert({
+          where: {
+            authProviderId: profile.id,
+          },
+          update: {
+            authProvider: strategy,
+            authProviderId: profile.id,
+            displayName: profile.displayName,
+          },
+          create: {
+            email: null,
+            displayName: profile.displayName,
+            authProvider: strategy,
+            authProviderId: profile.id,
+          },
+        });
+        return cb(null, user);
+      } catch (err) {
+        return cb(err);
+      }
+    }
+    try {
+      const user = await prisma.user.upsert({
+        where: {
+          email: email,
+        },
+        update: {
+          authProvider: strategy,
+          authProviderId: profile.id,
+          displayName: profile.displayName,
+        },
+        create: {
+          email: email,
+          displayName: profile.displayName,
+          authProvider: strategy,
+          authProviderId: profile.id,
+        },
+      });
+      return cb(null, user);
+    } catch (err) {
+      try {
+        // Existing user currently stored without email now returning with email
+        // Edge case - handled here
+        const user = await prisma.user.update({
+          where: {
+            authProviderId: profile.id,
+          },
+          data: {
+            email: email,
+            displayName: profile.displayName,
+            authProvider: strategy,
+            authProviderId: profile.id,
+          },
+        });
+        return cb(null, user);
+      } catch (err) {
+        return cb(err);
+      }
+    }
+  };
+};
+
+const callback = (provider: string) => {
+  return (req: Request, res: Response, next: NextFunction) => {
+    passport.authenticate(provider, { session: false }, (err, user, info) => {
+      if (err || !user) {
+        return res.redirect('/' + '?error=' + err?.message);
+      }
+      req.login(user, { session: false }, (err) => {
+        if (err) {
+          console.error(err);
+          return res.status(400).send({ err: err?.message || err });
+        }
+        try {
+          const payload = {
+            iss: PUBLIC_URL,
+            sub: user.id,
+            email: user.email || undefined,
+          };
+          const token = sign(payload, process.env.JWT_SECRET as string, {
+            expiresIn: process.env.JWT_EXPIRES_IN || '7d',
+          });
+          const cookiePayload = { token };
+          const expiresInDate = new Date(
+            Date.now() + ms(process.env.JWT_EXPIRES_IN || '7d')
+          );
+          res.cookie('auth', JSON.stringify(cookiePayload), {
+            domain: process.env.DOMAIN_NAME,
+            path: '/api',
+            secure: process.env.NODE_ENV === 'production',
+            httpOnly: true,
+            sameSite: 'strict',
```

Comment:
Just try none for development.

```suggestion
            sameSite: 'none',
```